aten/src/ATen/native/native_functions.yaml

```diff
@@ -3302,6 +3302,10 @@
   dispatch:
     CompositeImplicitAutograd: rms_norm_symint
 
+- func: _rms_norm_fused(Tensor input, int normalized_shape_ndim, Tensor weight, float eps) -> Tensor
```

Comment:
tbh I think you're making your life more complex for no good reason here.
You can set a MPS dispatch on rms_norm above, define a backward formula for it and call it a day.